test/Interop/Cxx/foreign-reference/Inputs/inheritance.h

```diff
@@ -284,7 +282,8 @@ __attribute__((swift_attr("retain:sameretain")))
 __attribute__((swift_attr("release:samerelease"))) B2 {};  // expected-error {{multiple functions 'sameretain' found; there must be exactly one retain function for reference type 'B2'}}
                                                             // expected-error@-1 {{multiple functions 'samerelease' found; there must be exactly one release function for reference type 'B2'}}
 
-struct D : B1, B2 {}; // expected-warning {{unable to infer SWIFT_SHARED_REFERENCE for 'D', although one of its transitive base types is marked as SWIFT_SHARED_REFERENCE}}
+struct D : B1, B2 {}; // expected-error {{multiple functions 'sameretain' found; there must be exactly one retain function for reference type 'D'}}
```

Comment:
I think it is fine that this warning is now promoted to error as the user can always add a SHARED_REFERENCE annotation to the derived class to disambiguate what retain/release functions needs to be called. So this diagnostic can be fixed by the users (via annotations or API notes).